SparkPost/2web2ui
# FE-1368 - Migrates tests cases from lengthy Cypress test suites to individual spec files, fixes test reporting to split tests by timing

### What Changed

- Fixes both Jest and Cypress test reporters so that Cypress tests can split between parallel runners based on their duration
- Test reports are now uploaded as artifacts to help with troubleshooting in the future
- Properly introduces the `.eslintplugin` directory in to the unit test step
- Upgrades `jest-junit` test reporter so that we can have access to the `"addFileAttribute"` configuration option.
- Uninstalls `cypress-multi-reporters` and `mocha-junit-reporter` in favor of `cypress-circleci-reporter` in order to have access to filenames
- Splits up tests within `/signals-analytics/analytics-report/tabs.spec.js` in to distinct specs for each type of table rendered within the tabs
- Splits up tests within `/domains/listPage.spec.js` in to distinct specs for each type of table rendered on that page
- Updates handling of cookies between tests to use `Cypress.Cookies.preserveOnce()` instead of the `preserve` option - allowing the login specs to properly clear cookies prior to running
- Removes video compression step for Cypress tests, saving even more time

### How To Test

- Verify that test reporter XML files are available as artifacts
- Verify that CircleCI is now picking up on test timing data - you can verify this by viewing the runner logs within CircleCI itself - let me know if you need help finding this
- Verify the pipeline runs significantly faster when compared with the default branch

### To Do
